apps/prairielearn/src/sprocs/sync_assessments.sql

```diff
@@ -450,45 +450,25 @@ BEGIN
                     RETURNING aq.id INTO new_assessment_question_id;
                     new_assessment_question_ids := array_append(new_assessment_question_ids, new_assessment_question_id);
 
+                    -- If the assessment is configured as group work, sync the role permissions.
                     IF (valid_assessment.data->>'group_work')::boolean THEN
-                        -- Iterate over all group roles in assessment
-                        FOR valid_group_role IN (
-                            SELECT gr.id, gr.role_name
-                            FROM group_roles as gr
-                            WHERE gr.assessment_id = new_assessment_id
-                        ) LOOP
-                            -- Insert roles that can view
-                            INSERT INTO assessment_question_role_permissions (
-                                assessment_question_id,
-                                group_role_id,
-                                can_view
-                            ) VALUES (
-                                new_assessment_question_id,
-                                valid_group_role.id,
-                                (valid_group_role.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_view')))
-                            ) ON CONFLICT (assessment_question_id, group_role_id)
-                            DO UPDATE
-                            SET
-                                assessment_question_id = EXCLUDED.assessment_question_id,
-                                group_role_id = EXCLUDED.group_role_id,
-                                can_view = EXCLUDED.can_view;
-
-                            -- Insert roles that can submit
-                            INSERT INTO assessment_question_role_permissions (
-                                assessment_question_id,
-                                group_role_id,
-                                can_submit
-                            ) VALUES (
-                                new_assessment_question_id,
-                                valid_group_role.id,
-                                (valid_group_role.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_submit')))
-                            ) ON CONFLICT (assessment_question_id, group_role_id)
-                            DO UPDATE
-                            SET
-                                assessment_question_id = EXCLUDED.assessment_question_id,
-                                group_role_id = EXCLUDED.group_role_id,
-                                can_submit = EXCLUDED.can_submit;
-                        END LOOP;
+                        INSERT INTO assessment_question_role_permissions (
+                            assessment_question_id,
+                            group_role_id,
+                            can_view,
+                            can_submit
```

Comment:
A minor behavior change: with the old code, `can_submit` would be left as `NULL` (the default) on new rows that only have view permission, but the new code will explicitly set it to `FALSE`. I think this is an improvement in any case, but I wanted to raise it in case it matters?

Reply:
I just checked all usages of `can_submit` and this doesn't make a difference. I agree that it's an improvement to always set explicitly to `FALSE`.